packages/apps/job-launcher/server/src/modules/payment/payment.service.ts

```diff
@@ -413,21 +410,40 @@ export class PaymentService {
     for (const token of uniqueTokens) {
       const tokenAmountSum = paymentEntities
         .filter((payment) => payment.currency === token)
-        .reduce((sum, payment) => sum + Number(payment.amount), 0);
+        .reduce((sum, payment) => add(sum, Number(payment.amount)), 0);
 
       const rate =
-        token === Currency.USD
+        token === FiatCurrency.USD
           ? 1
-          : await this.rateService.getRate(token, Currency.USD);
-
-      totalBalance += tokenAmountSum * rate;
+          : await this.rateService.getRate(token, FiatCurrency.USD);
+      totalBalance += mul(tokenAmountSum, rate);
     }
 
     return totalBalance;
   }
 
-  public async createRefundPayment(dto: PaymentRefundCreateDto) {
-    const rate = await this.rateService.getRate(TokenId.HMT, Currency.USD);
+  public async getUserBalanceByCurrency(
```

Comment:
I think that we're not going to support anything except for usd in the nearest future so we can remove this method and simplify a bit our business logic.

Reply:
Also we'll be able to get of paymentCurrencyRate etc.